### PR TITLE
update(ChatBar): Fix type letter render all time

### LINKED ISSUE #563: UI - Fix input lag
**Issue:** 
When you are typing in chat at a reasonable speed, the input will be delayed and really laggy. Usually running `--release` will fix that 

**OS:**
Windows 
Linux 
MacOS

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Now, when user types, it just render chatbar and not entire compose file 
a. States were changed to inner_state, a good approach to use in other parts of the app 
b. Was possible to remove input variable as well, and some others unnecessary code 
c. It seems, was possible to avoid re render own chatbat when typing as well

https://user-images.githubusercontent.com/63157656/229565108-34b44fdc-6fa6-4fb9-890e-d885a161eb9d.mov

#### 2.Typed as fast as possible to show functionality 

https://user-images.githubusercontent.com/63157656/229566649-5f2f2a17-e80f-4bd8-b830-ad637ae219d0.mov

- 

### Which issue(s) this PR fixes 🔨

- Resolve #563

### CODE REVIEW

**Review by sdwoodbury — APPROVED** [review]
nice and simple :)

File: ui/src/components/chat/compose.rs [hunk 1]
```diff
@@ -1096,9 +1084,16 @@ fn get_chatbar<'a>(cx: &'a Scoped<'a, ComposeProps>) -> Element<'a> {
...
         local_typing_ch.send(TypingIndicator::NotTyping);
 
-        let msg = input.read().clone();
-        // clearing input here should prevent the possibility to double send a message if enter is pressed twice
-        input.write().clear();
+        let msg = state
+            .read()
+            .get_active_chat()
+            .as_ref()
+            .and_then(|d| d.draft.clone())
```
Comment (dariusc93) [inline ui/src/components/chat/compose.rs:1091]: could probably clean it up by changing it to a `map` and performing the iteration here, but not a blocker.

**Review by dariusc93 — APPROVED** [review]
LGTM. Did leave a suggestion but not blocking

**Review by phillsatellite — APPROVED** [review]

**Comment by phillsatellite** [comment]
@lgmarchi I left Do Not Merge in case you wanted to do Darius's suggestions

File: ui/src/components/chat/compose.rs (hunk 1 above)
Comment (lgmarchi) [reply ui/src/components/chat/compose.rs:1091]: I didn't see so much difference my friend, long code using map as well

**Pull request merged**